score/mces.py

```diff
@@ -82,12 +82,14 @@ def initial_node_correspondences(graph1, graph2, identities1=None, identities2=N
                 # also determine the maximum number of edge matches we
                 # can hope to score, for each node-node correspondence
                 #
-                for edge1 in graph1.edges:
-                    for edge2 in graph2.edges:
-                        if edge1.lab == edge2.lab and \
-                           (edge1.src == node1.id and edge2.src == node2.id or
-                            edge1.tgt == node1.id and edge2.tgt == node2.id):
-                            edges[i, j] += 1;
+                src_edges1 = [ e for e in graph1.edges if e.src == node1.id ]
+                src_edges2 = [ e for e in graph2.edges if e.src == node2.id ]
+                tgt_edges1 = [ e for e in graph1.edges if e.tgt == node1.id ]
+                tgt_edges2 = [ e for e in graph2.edges if e.tgt == node2.id ]
+                src_edges_x = [ len([ 1 for e2 in src_edges2 if e1.lab == e2.lab]) for e1 in src_edges1]
```

Comment:
Wouldn't `sum(1 for e2 in src_edges2 if e1.lab == e2.lab)` be better than `len([ 1 for e2 in src_edges2 if e1.lab == e2.lab])`, for example? You don't have to create an actual list to calculate a sum. Same for the top-level sums.

Reply:
Thank you for your reviewing!  From my small experiment, "len" seems to be faster than "sum".

sample codes:
```python
def calc_time(f):
        s = time.clock()
        f()
        e = time.clock()
     return e - s

def cumsum(L):
    def f():
        for _ in range(L):
            sum(1 for _ in range(L))
     return f

def cumlen(L):
    def f():
        for _ in range(L):
            len([1 for _ in range(L)])
     return f
```

and the results:
```
In [23]: calc_time(cumsum(10000))
Out[23]: 5.929826000000002

In [25]: calc_time(cumlen(10000))
Out[25]: 3.4818
```

Reply:
Interesting, thanks for checking this!

Reply:
This is because the `sum(1 for _ in range(L))` expression internally creates a L length list and summaries over all 
 the list elements with O(L) order calculation, but `len([1 for _ in range(L)])` explicitly create a L length list and accessing "length" property of the list with O(1) order calculation.